### PR TITLE
ARGV: cache the results

### PR DESCRIPTION
**Pull request by xu-cheng**

### CODE REVIEW

File: Library/Homebrew/extend/ARGV.rb [hunk 1]
```diff
@@ -93,82 +93,100 @@ def value(arg)
   end
 
   def force?
-    flag? "--force"
+    return @force if instance_variable_defined?(:@force)
+    @force = flag? "--force"
```
Comment (bfontaine) [inline Library/Homebrew/extend/ARGV.rb]: Why didn’t you use `@force ||= flag? "--force"` here?

**Comment by bfontaine** [comment]
Good idea!

File: Library/Homebrew/extend/ARGV.rb (hunk 1 above)
Comment (xu-cheng) [inline Library/Homebrew/extend/ARGV.rb]: Because that logic cannot cache false result.

Comment (bfontaine) [inline Library/Homebrew/extend/ARGV.rb]: That’s right.

**Comment by mikemcquaid** [comment]
Do you have any before/after benchmarks for this? I'm struggling to believe this is a performance bottleneck.

**Comment by xu-cheng** [comment]
Yes it won't make any huge performance improvement. I do this to simplify code writing, so we can just have `ARGV.foo? ... ARGV.foo?` instead of `foo = AGRV.foo? ... <use foo afterwards>`.

**Comment by mikemcquaid** [comment]
I think it makes the code much harder to read. I also don't think you need to cache `ARGV` usage like that given it's just an array lookup.

**Pull request closed without merging**